### ISSUE TITLE
MediaPicker treats file as folder

### ISSUE THREAD

**Issue by roeermose** [umbraco/Umbraco-CMS#6441]
It is not possible to select the media in the mediapicker when using belowconfiguration.
The media is treated as a folder eventhough it is a file.

### Specifics
Version 7.15.3
Chrome 76.0.3809.132
Checked in Firefox 67.0.4

Have been tested on a clean 7.15.3 by support with same results.

### Steps to reproduce

Create a doc-type, create property with MediaPicker editor, set "Pick multiple items" = true, "Pick only images" = false, "Disable folder select" = true
Create a media where you do not upload a file.
Create a node using the new doc-type and try to select the media.

### Expected result
I expected to be able to select the media.

### Actual result
The media was not selectable, and when you tried to select it, then you would be presented with an empty folder.

---
_This item has been added to our backlog [AB#7087](https://dev.azure.com/umbraco/243e7927-03b2-44e2-908f-d4ac7ea5daaa/_workitems/edit/7087)_

**Comment by nul800sebastiaan** [comment]
Hey @roeermose - apart from the selected file not really having a file extension, this all works for me, not sure what I'm doing differently?

![image](https://user-images.githubusercontent.com/304656/65744349-6440f280-e0f8-11e9-9627-d0df6227b9c8.png)

**Comment by roeermose** [comment]
Hey @nul800sebastiaan - have you set the specfic configuration for the MediaPicker (default works fine, but the specific configuration does not)

**Comment by nul800sebastiaan** [comment]
Aha! Yeah, I did misread the configuration necessary, after updating it, I can't pick the empty item either. 

We'll have a look!

**Comment by stevemegson** [comment]
In 7.15, the media picker's definition of "is a folder" is effectively just "has no umbracoFile value". The current test can't tell the difference between an item which has no `umbracoFile` property and one where the `umbracoFile` property has no file selected. Both will behave as folders.

I'd need to check, but my PR #6195 to support thumbnails from other properties may also handle having no file selected. I think it will make mediaPath null when there's no property and an empty string when the property has no selected file.

There may also be a javascript tweak needed make sure that an empty string as mediaPath is enough to make the item "not a folder".

**Comment by andersbrohall** [comment]
Is this related?
[#7317](https://github.com/umbraco/Umbraco-CMS/issues/7317?fbclid=IwAR0ctuDwA7JfBwqfgd7EAy1S8sovKQq0OG_KMm0Jm85qr4NiP7GOLKPumjg)
:)

**Comment by banarualexandru** [comment]
Prerequisites: I'm using BlobStorage from Azure for storing the media files.

My findings regarding this issue.
**Problem 1**
After update from 7.12 to 7.15.3 The `cmsMedia` table is missing all the files. Just some were migrated after the normal migration.
**Solution**
- Removed all data from `cmsMedia` table
- Used part of this script, from [Issue-5848](https://github.com/umbraco/Umbraco-CMS/issues/5848#issuecomment-510801892), adapted to fill the `mediaPath` column as well. Final script:
`
DECLARE @i int
DECLARE @nodeId int
DECLARE @numrows int
DECLARE @versionId varchar(50)
DECLARE @mediaPath varchar(1000)
DECLARE @rootNodes TABLE (
    idx smallint Primary Key IDENTITY(1,1)
    , NodeId int
)

declare @mediaPaths Table(
	ids int Primary Key Identity(1,1)
	,mediaPath nvarchar(1000)
	,nodeId int
)

INSERT @mediaPaths 
SELECT cmsPropertyData.dataNvarchar as mediaPath, umbracoNode.id as nodeId
                            FROM cmsPropertyData
                            INNER JOIN cmsPropertyType ON cmsPropertyType.id = cmsPropertyData.propertytypeid
                            INNER JOIN umbracoNode ON umbracoNode.id = cmsPropertyData.contentNodeId
                            WHERE cmsPropertyType.Alias = ('umbracoFile')
                                AND umbracoNode.id IN (SELECT umbracoNode.id
                                    FROM umbracoNode 
		                            INNER JOIN cmsContent ON cmsContent.nodeId = umbracoNode.id
		                            INNER JOIN cmsContentType ON cmsContentType.nodeId = cmsContent.contentType
		                            INNER JOIN cmsPropertyType ON cmsPropertyType.contentTypeId = cmsContentType.nodeId
		                            WHERE cmsPropertyType.Alias = ('umbracoFile') AND umbracoNode.nodeObjectType = ('B796F64C-1F99-4FFB-B886-4BF4BC011A9C'))
                                AND (cmsPropertyData.dataNvarchar IS NOT NULL OR cmsPropertyData.dataNtext IS NOT NULL) 

INSERT @rootNodes
SELECT id FROM umbracoNode WHERE nodeObjectType = 'B796F64C-1F99-4FFB-B886-4BF4BC011A9C' AND id NOT IN (SELECT nodeId FROM cmsMedia)

SET @i = 1
SET @numrows = (SELECT COUNT(*) FROM @rootNodes)
IF @numrows > 0
    WHILE (@i <= (SELECT MAX(idx) FROM @rootNodes))
    BEGIN

        SET @nodeId = (SELECT NodeId FROM @rootNodes WHERE idx = @i)
		SET @mediaPath = (SELECT mediaPath FROM @mediaPaths WHERE nodeId = @nodeId)
        --Do something with Id here
        SET @versionId = (SELECT TOP 1 VersionId FROM cmsContentVersion WHERE ContentId = @nodeId ORDER BY VersionDate DESC)
		PRINT(@versionId)
		PRINT(@nodeId)
		INSERT INTO cmsMedia(nodeId, versionId, mediaPath) VALUES(@nodeId, @versionId, @mediaPath)
		
        SET @i = @i + 1
    END
	`
 When [this PR](https://github.com/umbraco/Umbraco-CMS/pull/6195) made by Stevemegson will be merged, this will be done automatically by Umbraco migrator.
@nul800sebastiaan are there any reasons why this PR wasn't merged to v7/dev branch?
This workaround fixes the existing media files, but the thumbnail is still not shown, as is analyzed [issue-5847](https://github.com/umbraco/Umbraco-CMS/issues/5847)
**Problem 2**
When uploading a new image in Media / Media picker it saves `NULL` on `mediaPath` column of  `cmsMedia`.
I didn't find a solution for this.
@nul800sebastiaan , @stevemegson  Any Ideas of some workarounds for this?
I was thinking to listen to `MediaService.Saved` event and save 'manually' that Azure URL to `cmsMedia` table.
Thanks

**Comment by mjbarlow** [comment]
I'm going to crosslink this issue with Courier https://github.com/umbraco/Umbraco.Courier.Issues/issues/14 as it is the same problem when couriering images.

"The media was not selectable, and when you tried to select it, then you would be presented with an empty folder."

**Comment by wakeup2k5** [comment]
I noticed this issue (thumbnails not showing, unable to pick media) after upgrading from 7.14.0 to 7.15.4. 

For us, I was able to narrow it down to an alternate media folder being configured.

I replicated the issue using a stock Umbraco 7.15.4 install (Visual Studio, NuGet install, Default Starter Kit).  Everything worked fine by default.  I was able to upload imagery, select it using a Media picker, etc.

Then I created a '/testmedia/' folder and modified the FileSystemProviders.config, as follows...

<Provider alias="media" type="Umbraco.Core.IO.PhysicalFileSystem, Umbraco.Core">
    <Parameters>
      <add key="virtualRoot" value=**"~/testmedia/"** />
    </Parameters>
</Provider>

Immediately after this, I was still able to upload imagery, but I was not able to select it.  It seemed to treat the newly uploaded imagery as a folder.

Would a fix for this (accounting for FileSystemProviders.config changes) be within scope of this existing issue, or should I create a new issue?

**Comment by stewartellis** [comment]
When is 7.15.5 coming? Would really like a fix from HQ instead of rolling our own. Thanks

**Comment by deMD** [comment]
This issue also seems to be present in Umbraco 8.5.5

**Comment by hfloyd** [comment]
I'm disappointed to see that this fix didn't make it into 7.15.5 as originally planned....

**Comment by stewartellis** [comment]
HQ -  a hotfix or quick turn around would be most welcome. This is a major issue.
Thank you

**Comment by RLuhse** [comment]
This particular issue is affecting a large number of people, so a quick fix would be very much appreciated. It looks as though it was first reported almost a year ago...

**Comment by ItagMichael** [comment]
Any confirmed hotfix for timeframe for 7.15.6 with this fix? This is affecting us significantly.

**Comment by jattwood** [comment]
I was waiting and waiting for this fix in 7.15.5, really wish it was a hot fix. This is impacting multiple clients and I don't have answers for them. Please make this a priority with thanks!

**Comment by GingerSquirrel** [comment]
If anyone is struggling with this error I put in a temporary fix of adding a file upload property to to my media item and giving it the alias 'umbracoFile' and making it required. The user just has to upload an empty bugfix.txt file to make it work.

It is definitely not a pretty fix but it is better than me hard-coding media into the site. You will also have to re-save all your current media for it to be effective on them.

**Comment by vipervf1** [comment]
Yes, we are still waiting on this bug fix as well.

**Comment by stewartellis** [comment]
A polite reminder to HQ - this bug affects a paid product - namely courier. A timely fix seems like it should be a done deal.

Thanks

**Comment by jrguay** [comment]
Hello!

I know @Shazwazza worked on this and mentioned needing further discussion. Is there an ETA for releasing? 

Thank you!

**Comment by nul800sebastiaan** [comment]
Just as a note, we're working on it in this sprint (ending the 28th). To set expectations: I am not sure when exactly we can do a release but it should hopefully be quite soon after the fix is made.

**Comment by Shazwazza** [comment]
Ready for review https://github.com/umbraco/Umbraco-CMS/pull/8520

The build output with artifacts will be here (once successfully built) https://dev.azure.com/umbraco/Umbraco%20Cms/_build/results?buildId=46124&view=results , we'll need people to test this out before it can be shipped so please let us know on the PR https://github.com/umbraco/Umbraco-CMS/pull/8520 what your results are.  Thanks!

**Comment by hfloyd** [comment]
@Shazwazza - It wasn't exactly clear on that page where to get a copy of the updated files for testing. The "Artifacts" option only shows 8.7 items.

**Comment by nul800sebastiaan** [comment]
@hfloyd .. Are you sure? Clicking on the files gives me 

![image](https://user-images.githubusercontent.com/304656/91418335-4b821500-e852-11ea-82de-baecc9c98f72.png)


Specifically the link would be: 
https://dev.azure.com/umbraco/Umbraco%20Cms/_build/results?buildId=46124&view=artifacts&type=publishedArtifacts

**Comment by hfloyd** [comment]
huh. Your direct link works. The original page I was looking at shows:

![image](https://user-images.githubusercontent.com/1252593/91751608-0aac3800-eb93-11ea-9ce1-6a24cf2d10d0.png)

and it wasn't clear where to click for the downloads.

Thanks

**Comment by nul800sebastiaan** [comment]
Ah yeah, they changed it recently from "artifacts" (which is a word that has a meaning) to an icon of a box saying "4 published" 🤷

**Comment by nul800sebastiaan** [comment]
Aha I never even saw the artifacts on the left hand side either haha! I understand the confusion.

They get bundled here now 

![image](https://user-images.githubusercontent.com/304656/91829085-47cf0380-ec41-11ea-9253-e1268abddd2d.png)

**Comment by hfloyd** [comment]
@nul800sebastiaan I wasn't able to get to this before, and now the Build seems to have disappeared. Can you direct me to the build and I will do a community review this week? Thanks!

**Comment by timeverts** [comment]
I believe I might be encountering a similar issue having just upgraded an Umbraco 7.6.0 site to Umbraco 7.15.7. I am also using Azure Blob storage for media items and both the media picker in the RTE field ('Link to Media') and the standalone media picker seem to have the problem where the custom media type is being treated as a folder rather than a file by the pickers.

@nul800sebastiaan, I see a fix is now slated for a [7.15.8 release](https://our.umbraco.com/download/releases/7158) which might still be a long way off. Do you have a hotfix for this issue in the mean time?

**Comment by drbldgtl** [comment]
Hi, just to report we also experience this issue when a media type has multiple Upload properties. Not sure if intended fix for this will also resolve this issue.

**Comment by Ankurtrue1** [comment]
Hi Guys,

I upgraded my Umbraco Version from 7.12.2 to 7.15.5.

However I am facing an similar issue, as mentioned in the above details, i.e. I cannot add the Media Picker images to the content. When I try to add an Image which is setup using an ImageCropper. The Media folder uploads it correctly, but cant use it to assign in the content. It just keeps on adding same like a folder.

Can someone please advise a workaround. I am just using simple file as source for Media

Below are the image for Media and the problem I have referred above.

![Content-Adding Image_Not_Working](https://user-images.githubusercontent.com/6064472/123625202-9416b000-d820-11eb-8a3e-afe7208ba5fd.JPG)
![ImageCropper-MediaUpload](https://user-images.githubusercontent.com/6064472/123625215-97aa3700-d820-11eb-9e8a-8f92ce9637f6.JPG)

Regards,
Ankur

**Comment by umbrabot** [comment]
Hiya @roeermose!

Thanks for reporting this issue. It might have been a while since you heard from us since a lot of things are happening at the moment.  

Right now, we're in the process of building Umbraco version 9 and we're now shifting our focus to exclusively work on v8 and v9. From now on Umbraco 7 is in security mode, which means we will no longer fix issues nor merge pull request unless they are severe security issues. You can read all about this in [our long term support and end of life policy](https://umbraco.com/products/knowledge-center/long-term-support-and-end-of-life/).  

This means that this issue will be closed and no longer looked at by the team. We realize this might be disappointing and we do apologize for not coming to a proper conclusion on this one. We hope you understand that we need to focus our active development somewhere and that means our efforts will go to the current and the future versions of Umbraco.

If it makes sense to look at this issue still for version 8, or even version 9, we'd love it if you reported it as a new issue so it can get the proper attention.

Thanks, from your friendly Umbraco GitHub bot 🤖 🙂

**Comment by marcoteodoro** [comment]
Hi @nul800sebastiaan / @Shazwazza, I'm I right to assume this was never merged into a release branch?
